src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Publish.targets

```diff
@@ -96,11 +96,12 @@
     <Delete Files="$(PublishDir)\$(TargetName)$(NativeBinaryExt)" />
     <Copy SourceFiles="$(NativeOutputPath)$(TargetName)$(NativeBinaryExt)" DestinationFolder="$(PublishDir)" />
 
-    <Delete Files="$(PublishDir)\$(TargetName).js" Condition="'$(_targetArchitecture)'=='wasm'"/>
-    <Copy SourceFiles="$(NativeOutputPath)$(TargetName).js" DestinationFolder="$(PublishDir)" Condition="'$(_targetArchitecture)'=='wasm'"/>
+    <Delete Files="$(PublishDir)\$(TargetName).js" Condition="'$(_targetOS)' == 'browser'"/>
+    <Copy SourceFiles="$(NativeOutputPath)$(TargetName).js" DestinationFolder="$(PublishDir)" Condition="'$(_targetOS)' == 'browser'"/>
 
-    <Delete Files="$(PublishDir)\$(TargetName).wasm" Condition="'$(_targetArchitecture)'=='wasm'"/>
-    <Copy SourceFiles="$(NativeOutputPath)$(TargetName).wasm" DestinationFolder="$(PublishDir)" Condition="'$(_targetArchitecture)'=='wasm'"/>
+    <!-- For Wasi, the .wasm file is the NativeBinaryExt -->
+    <Delete Files="$(PublishDir)\$(TargetName).wasm" Condition="'$(_targetOS)' == 'browser'"/>
+    <Copy SourceFiles="$(NativeOutputPath)$(TargetName).wasm" DestinationFolder="$(PublishDir)" Condition="'$(_targetOS)' == 'browser'"/>
```

Comment:
This is confusing: comment mentions WASI, but code does `"'$(_targetOS)' == 'browser'"`.

Reply:
Yes, don't think we need these lines at all

Reply:
Removed now